plugin.video.svtplay/default.py

```diff
@@ -0,0 +1,308 @@
+
+def view_category(genre):
+  programs = svt.getProgramsForGenre(genre)
+  if not programs:
+    return
+  for program in programs:
+    mode = MODE_PROGRAM
+    if program["type"] == "video":
+      mode = MODE_VIDEO
+    __create_dir_item(program, mode)
+
+def view_episodes(url):
+  """
+  Displays the episodes for a program
+  """
+  logging.log("View episodes for {}".format(url))
+  episodes = svt.getEpisodes(url.split("/")[-1])
+  if episodes is None:
+    logging.log("No episodes found")
+    return
+  for episode in episodes:
+    __create_dir_item(episode, MODE_VIDEO)
+
+def __add_clip_dir_item(url):
+  """
+  Adds the "Clips" directory item to a program listing.
+  """
+  params = {}
+  params["mode"] = MODE_CLIPS
+  params["url"] = url
+  __add_directory_item(localize(30108), params)
+
+def view_clips(url):
+  """
+  Displays the latest clips for a program
+  """
+  logging.log("View clips for {}".format(url))
+  clips = svt.getClips(url.split("/")[-1])
+  if not clips:
+    logging.log("No clips found")
+    return
+  for clip in clips:
+    __create_dir_item(clip, MODE_VIDEO)
+
+def view_search():
+  keyword = helper.getInputFromKeyboard(localize(30102))
+  if keyword == "" or not keyword:
+    view_start()
+    return
+  keyword = quote(keyword)
+  logging.log("Search string: " + keyword)
+  keyword = re.sub(r" ", "+", keyword)
+  keyword = keyword.strip()
+  results = svt.getSearchResults(keyword)
+  for result in results:
+    mode = MODE_VIDEO
+    if result["type"] == "program":
+      mode = MODE_PROGRAM
+    __create_dir_item(result["item"], mode)
+
+def __create_dir_item(article, mode):
+  """
+  Given an article and a mode; create directory item
+  for the article.
+  """
+  params = {}
+  params["mode"] = mode
+  params["url"] = article["url"]
+  folder = False
+  if mode == MODE_PROGRAM:
+    folder = True
+  info = None
+  if "info" in list(article.keys()):
+    if "onlyAvailableInSweden" in article["info"] and \
+        article["info"]["onlyAvailableInSweden"] and \
+        helper.getSetting(S_HIDE_RESTRICTED_TO_SWEDEN):
+      # Do not show geo restricted items
+      return
+    info = article["info"]
+  __add_directory_item(article["title"], params, article["thumbnail"], folder, False, info)
+
+def __add_next_page_item(next_page, section):
+  __add_directory_item("Next page",
+                   {"page": next_page,
+                    "mode": section})
+
+def start_video(video_id):
+  video_json = svt.getVideoJSON(video_id)
+  if video_json is None:
+    logging.log("ERROR: Could not get video JSON")
+    return
+  try:
+    show_obj = helper.resolveShowJSON(video_json)
+  except ValueError:
+    logging.log("Could not decode JSON for "+video_id)
+    return
+  if show_obj["videoUrl"]:
+    play_video(show_obj)
+  else:
+    dialog = xbmcgui.Dialog()
+    dialog.ok("SVT Play", localize(30100))
+
+def play_video(show_obj):
+  player = xbmc.Player()
+  start_time = time.time()
```

Comment:
Our API could well be improve to allow setting this without waiting for the player to start

Reply:
I honestly don't know why this is needed, I inherited the code from the orignal developer.

But if this is something all video addon devs have to do then I agree with you that the API should be updated to simplify this action.